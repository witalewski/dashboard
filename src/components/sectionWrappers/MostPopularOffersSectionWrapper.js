import React from 'react';
import styled from '@emotion/styled';
import { Section } from '../Section';

const StyledMostPopularOffersSectionWrapper = styled.div`
  min-width: 400px;
  .section-content {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }
`;

export const MostPopularOffersSectionWrapper = ({ className, data }) => (
  <StyledMostPopularOffersSectionWrapper className={className}>
    <Section title="Most popular offers" className="most-popular-offers">
      <div className="section-content">SECTION CONTENT</div>
    </Section>
  </StyledMostPopularOffersSectionWrapper>
);
