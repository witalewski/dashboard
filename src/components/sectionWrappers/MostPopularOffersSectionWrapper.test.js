import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import mockChartData from '../../fixtures/mockChartData.json';
import { MostPopularOffersSectionWrapper } from './MostPopularOffersSectionWrapper';

describe('MostPopularOffersSectionWrapper', () => {
  it('matches snapshot with mock data', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <MostPopularOffersSectionWrapper data={mockChartData.mostPopuar.data} />
    );
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
