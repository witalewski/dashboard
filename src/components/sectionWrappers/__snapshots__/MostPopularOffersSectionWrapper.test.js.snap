// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MostPopularOffersSectionWrapper matches snapshot with mock data 1`] = `
<Section
  title="Most popular offers"
>
  <PieChart
    barCategoryGap="10%"
    barGap={4}
    cx="50%"
    cy="50%"
    endAngle={360}
    height={250}
    innerRadius={0}
    layout="centric"
    margin={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      }
    }
    outerRadius="80%"
    reverseStackOrder={false}
    stackOffset="none"
    startAngle={0}
    width={350}
  >
    <Pie
      animationBegin={400}
      animationDuration={1500}
      animationEasing="ease"
      blendStroke={false}
      cx={120}
      cy={120}
      data={
        Array [
          Object {
            "name": "E946347906_AU",
            "value": 36,
          },
          Object {
            "name": "E643516498_AU",
            "value": 4,
          },
          Object {
            "name": "E657734674_AU",
            "value": 17,
          },
          Object {
            "name": "E135478956_AU",
            "value": 44,
          },
        ]
      }
      dataKey="value"
      endAngle={360}
      fill="#8884d8"
      hide={false}
      innerRadius={40}
      isAnimationActive={true}
      label={true}
      labelLine={true}
      legendType="rect"
      minAngle={0}
      nameKey="name"
      onAnimationEnd={[Function]}
      outerRadius={80}
      paddingAngle={0}
      startAngle={0}
      stroke="#fff"
    >
      <Cell
        fill="#5db990"
      />
      <Cell
        fill="#558ca0"
      />
      <Cell
        fill="#864ba5"
      />
      <Cell
        fill="#829ac0"
      />
    </Pie>
    <Tooltip
      active={false}
      animationDuration={400}
      animationEasing="ease"
      contentStyle={Object {}}
      coordinate={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      cursor={true}
      cursorStyle={Object {}}
      filterNull={true}
      isAnimationActive={true}
      itemSorter={[Function]}
      itemStyle={Object {}}
      labelStyle={Object {}}
      offset={10}
      separator=" : "
      useTranslate3d={false}
      viewBox={
        Object {
          "x1": 0,
          "x2": 0,
          "y1": 0,
          "y2": 0,
        }
      }
      wrapperStyle={Object {}}
    />
    <Legend
      align="right"
      iconSize={14}
      layout="vertical"
      verticalAlign="bottom"
      width={200}
      wrapperStyle={
        Object {
          "left": 250,
          "top": 80,
        }
      }
    />
  </PieChart>
</Section>
`;
